applications/ShapeOptimizationApplication/python_scripts/communicator_factory.py

```diff
@@ -29,92 +29,98 @@ def __init__( self, optimization_settings ):
         self.__initializeListOfResponses()
 
     # --------------------------------------------------------------------------
-    def initializeCommunication( self ):
+    def initializeCommunication(self):
         self.__deleteAllRequests()
         self.__deleteAllReportedValues()
 
     # --------------------------------------------------------------------------
-    def requestValueOf( self, response_id ):
+    def requestValueOf(self, response_id):
         self.list_of_requests[response_id]["calculateValue"] = True
 
     # --------------------------------------------------------------------------
-    def requestGradientOf( self, response_id ):
+    def requestGradientOf(self, response_id):
         self.list_of_requests[response_id]["calculateGradient"] = True
 
     # --------------------------------------------------------------------------
-    def isRequestingValueOf( self, response_id ):
+    def isRequestingValueOf(self, response_id):
         if response_id not in self.list_of_requests.keys():
             return False
         else:
             return self.list_of_requests[response_id]["calculateValue"]
 
     # --------------------------------------------------------------------------
-    def isRequestingGradientOf( self, response_id ):
+    def isRequestingGradientOf(self, response_id):
         if response_id not in self.list_of_requests.keys():
             return False
         else:
             return self.list_of_requests[response_id]["calculateGradient"]
 
     # --------------------------------------------------------------------------
-    def reportValue( self, response_id, value ):
-        self.__storeValue( response_id, value)
-        if self.__isResponseWaitingForInitialValueAsReference( response_id ):
-            self.__setValueAsReference( response_id, value )
+    def reportValue(self, response_id, value):
+        self.__storeValue(response_id, value)
+        if self.__isResponseWaitingForInitialValueAsReference(response_id):
+            self.__setValueAsReference(response_id, value)
 
-        standardized_value = self.__translateValueToStandardForm( response_id, value )
-        self.__storeStandardizedValue( response_id, standardized_value )
+        standardized_value = self.__translateValueToStandardForm(response_id, value)
+        self.__storeStandardizedValue(response_id, standardized_value)
 
     # --------------------------------------------------------------------------
-    def reportGradient( self, response_id, gradient ):
-        standardized_gradient = self.__translateGradientToStandardForm( response_id, gradient )
-        self.__storeStandardizedGradient( response_id, standardized_gradient )
+    def reportGradient(self, response_id, gradient):
+        dimension = len(list(gradient.values())[0])
```

Comment:
`next(iter(dictionary.values()))` should be more efficient.
Also a check needs to be done if it is empty...